contracts/components/scanners/ScannerToNodeRunnerMigration.sol

```diff
@@ -0,0 +1,156 @@
+     * @param __nodeRunnerRegistry address of NodeRunnerRegistry (new registry)
+     * @param __migrationEndTime time when migration period ends 
+     */
+    function initialize(
+        address __manager,
+        address __scannerNodeRegistry,
+        address __nodeRunnerRegistry,
+        uint256 __migrationEndTime
+    ) public initializer {
+        __BaseComponentUpgradeable_init(__manager);
+
+        _setScannerNodeRegistry(__scannerNodeRegistry);
+        _setNodeRunnerRegistry(__nodeRunnerRegistry);
+        _setMigrationEndTime(__migrationEndTime);
+    }
+
+    /**
+     * @notice Method to self migrate from the old ScannerRegistry NFTs to a single NodeRunnerRegistry NFT.
+     * WARNING: ScannerNodeRegistry's manager addresses will not be migrated, please user NodeRunnerRegistry's methods to set them again.
+     * @param scanners array of scanner addresses to be migrated. Have to be less than MAX_SCANNERS.
+     * All the enabled (disabled flags set to 0) ScannerRegistry ERC721 identified by the uint256(address) in the input array will be:
+     * - Registered in NodeRunnerRegistry to the nodeRunnerId either indicated or generated, with the same chainId and metadata.
+     * - Deleted in ScannerNodeRegistry. The ERC721 will be burned, disabled flags and managers deleted from storage.
+     * Scanners with disabled flags != 0 will be ignored (opted out), and will stay disabled in ScannerNodeRegistry.
+     * @param nodeRunnerId If the set as 0, the NodeRunnerRegistry ERC721 will be minted to the sender (address must not own any),
+     * set as a valid NodeRunnerRegistry ERC721 id owned by sender address otherwise.
+     * @return NodeRunnerRegistry ERC721 id the scanners are migrated to.
+     */
+    function selfMigrate(address[] calldata scanners, uint256 nodeRunnerId) external returns (uint256) {
+        return _migrate(scanners, nodeRunnerId, _msgSender());
+    }
+
+    /**
+     * @notice Method to migrate from the old ScannerRegistry NFTs to a single NodeRunnerRegistry NFT, executed by an address with the role
+     * MIGRATION_EXECUTOR_ROLE.
+     * WARNING: ScannerNodeRegistry's manager addresses will not be migrated, please user NodeRunnerRegistry's methods to set them again.
+     * @param scanners array of scanner addresses to be migrated. Have to be less than MAX_SCANNERS.
+     * All the enabled (disabled flags set to 0) ScannerRegistry ERC721 identified by the uint256(address) in the input array will be:
+     * - Registered in NodeRunnerRegistry to the nodeRunnerId either indicated or generated, with the same chainId and metadata.
+     * - Deleted in ScannerNodeRegistry. The ERC721 will be burned, disabled flags and managers deleted from storage.
+     * Scanners with disabled flags != 0 will be ignored (opted out), and will stay disabled in ScannerNodeRegistry.
+     * @param nodeRunnerId If the set as 0, the NodeRunnerRegistry ERC721 will be minted to nodeRunner (address must not own any),
```

Comment:
What's the reason we require the address doesn't own a nodeRunner token when they request a new token id?

Reply:
The idea was to have all the Scanners be managed by one NodeRunner NFT, but it is not true anymore in https://github.com/forta-network/forta-contracts/pull/130 because we need to have (at least) one NodeRunner NFT per chain, so I'm removing the check.

I'll let it stay here since it will be changed next